llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/Passes/Passes.h

```diff
@@ -90,7 +90,7 @@ std::unique_ptr<OperationPass<func::FuncOp>> createLinalgExtToLoopsPass();
 /// Container of information needed to materialize the pack operation.
 struct MaterializeEncodingInfo {
   SmallVector<int64_t> innerDimsPos;
-  SmallVector<int64_t> innerTileSizes;
+  SmallVector<OpFoldResult> innerTileSizes;
```

Comment:
Just a note, this can be quite a footgun. If a `Value` is used here, based on where the `Value` is created and where it is used it is very easy to get into SSA use before def violations.... There are other options we can consider that are more robust. Most robust would be using a symbol and the symbol could have a potentially static value which can be used when needed. So this could be made a `SymbolRef`.
If we use a `OpFoldResult` we will need to make sure the value is created at the very top of the function to avoid SSA use-def issues.

Reply:
I would like to be correct and non footgun-ridden! I have looked at `SymbolRefAttr` but did not find example usage making it obvious how to wrap a Value into that , can you share some pointer?

Also, can you elaborate on what is so special about the present situation that we can't have a `Value` in `MaterializeEncodingInfo` ?